### DISCUSSION TITLE
MASTER-SLAVE COMMUNICATION VIA I2C

### DISCUSSION THREAD

**Discussion by pacolop** [tttapa/Control-Surface#831] [General]
Hello,
I tell you a question I'm getting stuck.
I have two pcb with arduino atmega 32u4 which are prepared for i2c communication.
In both I have an 8*6 matrix that works perfectly (each plate has a different channel in the matrix)

What I need is help to be able to connect both boards via i2c and for the slave to send the notes emitted to the master when we press the buttons, the master simply has to listen and transfer via USB what the master and its own matrix sends.

I would really appreciate the help

**Comment by Veroledez** [comment]
Hola, ya que tienes dos leonardos,    mejor aprovecha el puerto serial extra para comunicarlos, y luego usa una tuberia midi para redireccionar    los mensajes del serial al usb
